x-pack/plugin/ml/src/test/java/org/elasticsearch/xpack/ml/action/TransportDeleteTrainedModelActionTests.java

```diff
@@ -0,0 +1,132 @@
+import org.elasticsearch.action.admin.cluster.node.tasks.cancel.CancelTasksRequestBuilder;
+import org.elasticsearch.action.admin.cluster.node.tasks.cancel.CancelTasksResponse;
+import org.elasticsearch.action.admin.cluster.node.tasks.list.ListTasksAction;
+import org.elasticsearch.action.admin.cluster.node.tasks.list.ListTasksResponse;
+import org.elasticsearch.action.support.PlainActionFuture;
+import org.elasticsearch.client.internal.Client;
+import org.elasticsearch.rest.RestStatus;
+import org.elasticsearch.test.ESTestCase;
+import org.elasticsearch.threadpool.TestThreadPool;
+import org.elasticsearch.threadpool.ThreadPool;
+import org.junit.After;
+import org.junit.Before;
+
+import java.util.Collections;
+
+import static org.elasticsearch.xpack.ml.action.TransportDeleteTrainedModelAction.cancelDownloadTask;
+import static org.elasticsearch.xpack.ml.utils.TaskRetrieverTests.getTaskInfoListOfOne;
+import static org.elasticsearch.xpack.ml.utils.TaskRetrieverTests.mockClientWithTasksResponse;
+import static org.elasticsearch.xpack.ml.utils.TaskRetrieverTests.mockListTasksClient;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.nullValue;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.same;
+import static org.mockito.Mockito.doAnswer;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class TransportDeleteTrainedModelActionTests extends ESTestCase {
+    private ThreadPool threadPool;
+
+    @Before
+    public void setUpThreadPool() {
+        threadPool = new TestThreadPool(getTestName());
+    }
+
+    @After
+    public void tearDownThreadPool() {
+        terminate(threadPool);
+    }
+
+    public void testCancelDownloadTaskCallsListenerWithNullWhenNoTasksExist() {
+        var client = mockClientWithTasksResponse(Collections.emptyList(), threadPool);
+        var listener = new PlainActionFuture<CancelTasksResponse>();
+
+        cancelDownloadTask(client, "modelId", listener);
+
+        assertThat(listener.actionGet(), nullValue());
+    }
+
+    public void testCancelDownloadTaskCallsOnFailureWithErrorWhenCancellingFailsWithAnError() {
+        var client = mockClientWithTasksResponse(getTaskInfoListOfOne(), threadPool);
+        mockCancelTask(client);
+
+        doAnswer(invocationOnMock -> {
+            @SuppressWarnings("unchecked")
+            ActionListener<CancelTasksResponse> listener = (ActionListener<CancelTasksResponse>) invocationOnMock.getArguments()[2];
+            listener.onFailure(new Exception("cancel error"));
+
+            return Void.TYPE;
+        }).when(client).execute(same(CancelTasksAction.INSTANCE), any(), any());
+
+        var listener = new PlainActionFuture<CancelTasksResponse>();
+
+        cancelDownloadTask(client, "modelId", listener);
+
+        var exception = expectThrows(ElasticsearchException.class, listener::actionGet);
```

Comment:
general comment: you heavily use `PlainActionFuture<...>`: when you call `actionGet` can it timeout?

As the tests run in CI we try to always have timeouts as we don't want to have a CI run that gets stuck, because a listener isn't called. We don't have to be super strict here (especially windows builds are very slow), we usually use `30s`, if that feels to long you can also use `5s` or `10s`. Just ensure that it can't block "forever".